core/sql/executor/ExFirstN.cpp

```diff
@@ -228,11 +232,11 @@ short ExFirstNTcb::work()
 		// GET_N request value and firstNRows_ to my child.
 		if ((pentry_down->downState.request != ex_queue::GET_N) ||
 		    (pentry_down->downState.requestValue == firstnTdb().firstNRows()))
-		  centry->downState.requestValue = firstNVal;
+		  centry->downState.requestValue = effectiveFirstN_;
```

Comment:
Actually, that made me realize that on new line 234 the code also needs to be changed to effectiveFirstN_.